docs/mp/persistence.adoc

```diff
@@ -0,0 +1,1407 @@
+* <<DS, Named Data Source Integration>>
+** <<DS-Setup, Project Setup>>
+*** <<DS-CP-Project-Setup, Setting Up a Connection Pool>>
+**** <<DS-HikariCP-Project-Setup, Setting Up the HikariCP Connection Pool>>
+***** <<DS-HikariCP-Maven-Coordinates, Maven Coordinates (HikariCP)>>
+**** <<DS-UCP-Project-Setup, Setting Up the Oracle Universal Connection Pool>>
+***** <<DS-UCP-Maven-Coordinates, Maven Coordinates (Oracle Universal Connection Pool)>>
+*** <<DS-Driver-Project-Setup, Setting Up a Database Driver>>
+**** <<DS-H2-Driver-Project-Setup, Setting Up H2>>
+***** <<DS-H2-Driver-Maven-Coordinates, Maven Coordinates (H2)>>
+**** <<DS-Oracle-Driver-Project-Setup, Setting Up Oracle JDBC>>
+***** <<DS-Oracle-Driver-Maven-Coordinates, Maven Coordinates (Oracle JDBC)>>
+** <<DS-Configuration, Configuration>>
+*** <<DS-Configuration-Examples, Examples>>
+** <<DS-Usage, Usage>>
+* <<JTA, Jakarta Transactions (JTA) Integration>>
+** <<JTA-Maven-Coordinates, Maven Coordinates>>
+** <<JTA-Configuration, Configuration>>
+** <<JTA-Usage, Usage>>
+* <<JPA, Jakarta Persistence (JPA) Integration>>
+** <<JPA-Setup, Project Setup>>
+*** <<JPA-Maven-Coordinates, Common Maven Coordinates>>
+*** <<JPA-MetaModel, Setting Up Static Metamodel Generation>>
+*** <<JPA-Hibernate-Maven-Coordinates, Maven Coordinates (Hibernate ORM)>>
+*** <<JPA-Hibernate-Weaving, Setting Up Static Weaving (Hibernate ORM)>>
+*** <<JPA-Eclipselink-Maven-Coordinates, Maven Coordinates (Eclipselink)>>
+*** <<JPA-Eclipselink-Weaving, Setting Up Static Weaving (Eclipselink)>>
+** <<JPA-Configuration, Configuration>>
+** <<JPA-Usage, Usage>>
+** <<JPA-Examples, Examples>>
+* <<References, References>>
+
+== Overview [[Overview]]
+
+Helidon MP comes with deep integration for three
+specification-defined, broadly persistence-related technologies that
+can be used together or separately:
+
+* <<DS, Named data sources>>
+* <<JTA, Jakarta Transactions (JTA)>>
+* <<JPA, Jakarta Persistence (JPA)>>
+
+Each integration's setup, configuration, and usage are described
+below.
+
+== Named Data Source Integration [[DS]]
+
+=== Overview
+
+Helidon MP's named data source integration allows you to safely inject
+managed
+link:{jdk-javadoc-url}/java.sql/javax/sql/DataSource.html[`javax.sql.DataSource`]
+instances that are annotated with
+link:{jakarta-inject-javadoc-url}/jakarta/inject/named[`jakarta.inject.Named`
+annotations] into your Helidon MP application.
+link:{jdk-javadoc-url}/java.sql/java/sql/Connection.html[`java.sql.Connection`
+objects]
+link:{jdk-javadoc-url}/java.sql/javax/sql/DataSource.html#getConnection()[acquired]
+from these data sources will be pooled by your choice of one of two
+possible connection pool implementations.
+
+The connections managed by the connection pool will be supplied by
+your relational database vendor's JDBC driver.
+
+How you set up Helidon MP's named data source integration differs
+depending on which of these two connection pools, which JDBC driver,
```

Comment:
depending on the connection pool, JDBC driver and relational database products you use.